src/workerd/api/http.h

```diff
@@ -393,6 +393,15 @@ class Fetcher: public jsg::Object {
   // `outgoingFactory` is used for Fetchers that use ad-hoc WorkerInterface instances, such as ones
   // created for Actors.
 
+  Fetcher(kj::Own<OutgoingFactory> outgoingFactory,
```

Comment:
I'm a little concerned that this means `OutgoingFactory` cannot be classified as either being a KJ I/O type or a JSG isolate heap type... it may be used in either context. Also concerning that if someone forgets to do `context.addObject()` they end up calling the wrong constructor which will be subtly broken but probably passing tests.

Can we introduce a new type like `CrossContextOutgoingFactory` that is meant to be a JSG type? We could even pass `IoChannelFactory&` as a parameter to `newSingleUseClient()` for this interface, to make clear that this can differ from call to call.

Alternatively I'm also happy with a variant of `Fetcher` that uses a capnp channel directly. (Though we'd have to figure out how to distinguish the channel type to the constructor, I guess...)